utils/loading_cache.hh

```diff
@@ -290,6 +290,34 @@ public:
         });
     }
 
+    value_ptr get_ptr(const Key& k, value_type value) {
+        // We shouldn't be here if caching is disabled
+        assert(caching_enabled());
+
+        timestamped_val_ptr ts_val_ptr = _loading_values.get_or_insert(k, std::move(value));
+        // check again since it could have already been inserted and initialized
+        if (!ts_val_ptr->ready() && !ts_val_ptr.orphaned()) {
+            _logger.trace("{}: storing the value for the first time", k);
+
+            if (ts_val_ptr->size() > _max_size) {
+                throw entry_is_too_big();
+            }
+
+            ts_value_lru_entry* new_lru_entry = Alloc().template allocate_object<ts_value_lru_entry>();
+
+            // Remove the least recently used items if map is too big.
+            shrink();
+
+            new(new_lru_entry) ts_value_lru_entry(std::move(ts_val_ptr), *this);
+
+            // This will "touch" the entry and add it to the LRU list - we must do this before the shrink() call.
+            value_ptr vp(new_lru_entry->timestamped_value_ptr());
```

Comment:
But it's after the shrink() call.

Reply:
I've actually reused the code from the original `get_ptr`. Considering that it works, probably only the comment needs changing - do we want to adjust it there as well?

Reply:
No idea.

Reply:
I don't understand what's the issue with the comment (or the original code) that needs to be fixed?

What do you, guys, don't understand exactly? 
@avikivity @wmitros Please, let me know and I'll clarify it to you.

Reply:
This comment says "we must do this **before** the `shrink()` call" even though it's a few lines **after** the `shrink()` call.
Perhaps it refers to some other `shrink`, but we can't deduce it from the comment

Reply:
@vladzcloudius does that explain the contradiction here? Also, are there still any parts of the patch that seem incorrect or are not justified enough?

Reply:
Ops. - Just noticed that I actually described all the above in the corresponding commit message:  1a9c6d9fd3af5b9dd06bc51128fdcd3185456c70

Reply:
> I've actually reused the code from the original `get_ptr`. Considering that it works, probably only the comment needs changing - do we want to adjust it there as well?

Maybe get_ptr() should call put() to avoid duplication.